test/metrics/spanner-metrics-exporter.ts

```diff
@@ -0,0 +1,227 @@
+      project_id: PROJECT_ID,
+      instance_id: INSTANCE_ID,
+      instance_config: 'test_config',
+      location: LOCATION,
+      client_hash: 'test_hash',
+      client_uid: 'test_uid',
+      client_name: 'test_name',
+      database: DATABASE_ID,
+      method: 'test_method',
+      status: 'test_status',
+      other: 'ignored',
+    };
+
+    attempt_counter = meter.createCounter(METRIC_NAME_ATTEMPT_COUNT, {
+      description: 'Count of attempts',
+      unit: '1',
+    });
+
+    operation_counter = meter.createCounter(METRIC_NAME_OPERATION_COUNT, {
+      description: 'Count of operations',
+      unit: '1',
+    });
+
+    gfe_connectivity_error_count = meter.createCounter(
+      METRIC_NAME_GFE_CONNECTIVITY_ERROR_COUNT,
+      {
+        description: 'Count of missing headers',
+        unit: '1',
+      },
+    );
+
+    attempt_latency = meter.createHistogram(METRIC_NAME_ATTEMPT_LATENCIES, {
+      description: 'Test attempt latencies in ms',
+      unit: 'ms',
+    });
+
+    operation_latency = meter.createHistogram(METRIC_NAME_OPERATION_LATENCIES, {
+      description: 'Test operation latencies in ms',
+      unit: 'ms',
+    });
+
+    gfe_latencies = meter.createHistogram(METRIC_NAME_GFE_LATENCIES, {
+      description: 'Test GFE latencies in ms',
+      unit: 'ms',
+    });
+  });
+
+  it('should export GCM metrics', async () => {
+    attempt_counter.add(10, metricAttributes);
+    operation_counter.add(25, metricAttributes);
+    gfe_connectivity_error_count.add(12, metricAttributes);
+    attempt_latency.record(30, metricAttributes);
+    operation_latency.record(45, metricAttributes);
+    gfe_latencies.record(22, metricAttributes);
+
+    const {errors, resourceMetrics} = await reader.collect();
+    if (errors.length !== 0) {
+      throw errors;
+    }
+
+    const sendTimeSeriesStub = sinon
+      .stub(exporter as any, '_sendTimeSeries')
+      .resolves();
+
+    await new Promise<ExportResult>(resolve => {
+      exporter.export(resourceMetrics, result => {
+        if (result.error) {
+          console.error(result.error);
+        }
+        resolve(result);
+      });
+    });
+
+    assert(sendTimeSeriesStub.calledOnce);
+
+    const [timeseries] = sendTimeSeriesStub.getCall(0).args;
+
+    assert.strictEqual(timeseries.length, 6);
+  });
+
+  it('should exit early if resource metrics are empty', async () => {
+    const {errors, resourceMetrics} = await reader.collect();
+
+    if (errors.length !== 0) {
+      throw errors;
+    }
+    const sendTimeSeriesStub = sinon
+      .stub(exporter as any, '_sendTimeSeries')
+      .resolves();
+
+    await new Promise<ExportResult>(resolve => {
+      exporter.export(resourceMetrics, result => {
+        if (result.error) {
+          console.error(result.error);
+        }
+        resolve(result);
+      });
+    });
+
+    assert(sendTimeSeriesStub.notCalled);
+  });
+
+  it('should handle failed send during time series export with callback', async () => {
```

Comment:
nit: can we also add a test case for the `max_batch_export_size` that createTimeSeries is called multiple times as per the batch size

Reply:
Added test for batches below.